docs/tutorials/setup-opentelemetry.md

```diff
@@ -46,23 +48,68 @@ const sdk = new NodeSDK({
 sdk.start();
 ```
 
-In the `index.ts`, import this file **at the beginning**:
+Your probably won't need all the instrumentation inside `getNodeAutoInstrumentations()` so make sure to
+check the [documentation](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node) and tweak it properly.
 
-```typescript
-import './instrumentation'; // Setup the OpenTelemetry instrumentation
+It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
 
-// other imports and backend init...
-```
+This is why we will use the nodejs [`--require`](https://nodejs.org/api/cli.html#-r---require-module)
+flag when we start up the application.
 
-It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
+In your `Dockerfile` add the `--require` flag which points to the `instrumentation.ts` file
+
+```Dockerfile
+FROM node:18-bookworm-slim
+...
+# More functionality goes here
+...
+WORKDIR /app
+RUN chown node:node /app
+USER node
+
+ENV NODE_ENV production
+
+COPY --chown=node:node .yarn ./.yarn
+COPY --chown=node:node .yarnrc.yml ./
+
+# We need the instrumentation file inside the Docker image so we can use it with --require
+// highlight-add-next-line
+COPY --chown=node:node packages/backend/instrumentation.ts ./
+
+COPY --chown=node:node yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
+
+RUN --mount=type=cache,target=/home/node/.yarn/berry/cache,sharing=locked,uid=1000,gid=1000 \
+    yarn workspaces focus --all --production
+
+COPY --chown=node:node packages/backend/dist/bundle.tar.gz app-config*.yaml ./
+RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
+
+// highlight-remove-next-line
+CMD ["node", "packages/backend", "--config", "app-config.yaml"]
+// highlight-add-next-line
+CMD ["node", "--require", "./instrumentation.ts", "packages/backend", "--config", "app-config.yaml"]
```

Comment:
Is this needed? When running it locally, loading the `instrumentation.ts` at the top of my index.ts seemed to work.

Reply:
The whole reason for this revise is because it doesn't perfectly when deployed. Yes, adding the instrumentation file will work locally, but once you deploy, the code will get bundled and that will break some of the opentelemetry functionalities. For the second comment, check that you don't have some .dockerignore filter blocking this

Reply:
Hmm my docker ignore looks _ok_, but I can see that the file is not preset in the docker container, so it must not be copied over for some reason 😕 
```
node_modules
packages/*/src
packages/*/node_modules
plugins
*.local.yaml
```

shouldnt match `packages/backend/instrumentation.ts`

Reply:
Yeah I mean looking at the dockerfile, i see no reason why this would be copied into the container

Reply:
Oh man, i missed this line 
```
COPY --chown=node:node packages/backend/instrumentation.ts ./
```
sorry about that!